File: android/src/main/java/com/steerpath/rnsmartmap/RNSmartMapPackage.java
package com.steerpath.rnsmartmap;

import android.app.Activity;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class RNSmartMapPackage implements ReactPackage {
    public RNSmartMapPackage(Activity activity) {
    } // backwards compatibility

    public RNSmartMapPackage() {}


    @Override
    public List<NativeModule> createNativeModules(
            ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();

        modules.add(new RNSmartMapManager(reactContext));
        modules.add(new RNSmartLocationManager(reactContext));
        return modules;
    }

    @NonNull
    @Override
    public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactApplicationContext) {
        return Collections.emptyList();
    }
}

File: android/src/main/java/com/steerpath/rnsmartmap/RNSmartMapPackage.kt
package com.steerpath.rnsmartmap

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

class RNSmartMapPackage : ReactPackage {

    // 1. Return type must be List<NativeModule> (No '?' and no 'Mutable')
    override fun createNativeModules(
        reactContext: ReactApplicationContext
    ): List<NativeModule> {
        val modules = mutableListOf<NativeModule>() // Create it as mutable locally

        modules.add(RNSmartMapManager(reactContext))
        modules.add(RNSmartLocationManager(reactContext))

        return modules // Kotlin automatically casts MutableList to List
    }

    // 2. Return type must be List<ViewManager<*, *>> (No '?' and no 'Mutable')
    override fun createViewManagers(
        reactContext: ReactApplicationContext
    ): List<ViewManager<*, *>> {
        return emptyList() // The most idiomatic way to return an empty list in Kotlin
    }
}